testsuite/tests/kuadrant/authorino/operator/tls/conftest.py

```diff
@@ -0,0 +1,119 @@
+
+
+@pytest.fixture(scope="session")
+def authorino_domain(openshift):
+    """
+    Hostname of the upstream certificate sent to be validated by APIcast
+    May be overwritten to configure different test cases
+    """
+    return f"*.{openshift.project}.svc.cluster.local"
+
+
+@pytest.fixture(scope="session")
+def envoy_authority(wildcard_domain, cfssl):
+    """Authority for all certificates that envoy should accept"""
+    return cfssl.create_authority("backend-ca", hosts=[wildcard_domain])
+
+
+@pytest.fixture(scope="session")
+def invalid_authority(wildcard_domain, cfssl):
+    """Completely unrelated CA for generating certificates which should not succeed"""
+    return cfssl.create_authority("server-ca", hosts=[wildcard_domain])
+
+
+@pytest.fixture(scope="session")
+def authorino_authority(authorino_domain, cfssl):
+    """Authority for Authorino Certificate"""
+    return cfssl.create_authority("authorino-ca", hosts=[authorino_domain])
+
+
+@pytest.fixture(scope="session")
+def authorino_cert(authorino_domain, cfssl, authorino_authority):
+    """Authorino certificate"""
+    return cfssl.create("authorino", hosts=[authorino_domain], certificate_authority=authorino_authority)
+
+
+@pytest.fixture(scope="session")
+def envoy_cert(wildcard_domain, cfssl, envoy_authority):
+    """Certificate that is actively used by Envoy"""
+    return cfssl.create("envoy", hosts=[wildcard_domain], certificate_authority=envoy_authority)
+
+
+@pytest.fixture(scope="session")
+def valid_cert(wildcard_domain, cfssl, envoy_authority):
+    """Certificate accepted by Envoy"""
+    return cfssl.create("valid", hosts=[wildcard_domain], certificate_authority=envoy_authority)
+
+
+@pytest.fixture(scope="session")
+def invalid_cert(invalid_authority, cfssl, wildcard_domain):
+    """Certificate rejected by Envoy"""
+    return cfssl.create("invalid", hosts=[wildcard_domain], certificate_authority=invalid_authority)
+
+
+@pytest.fixture(scope="module")
+def envoy(request, authorino, openshift, create_secret, blame, label, backend,
+          authorino_authority, envoy_authority, envoy_cert):
+    """Envoy + Httpbin backend"""
+    authorino_secret = create_secret(authorino_authority, "authca")
+    envoy_ca_secret = create_secret(envoy_authority, "backendca")
+    envoy_secret = create_secret(envoy_cert, "envoycert")
+
+    envoy = TLSEnvoy(openshift, authorino, blame("backend"), label, backend.url,
+                     authorino_secret, envoy_ca_secret, envoy_secret)
+    request.addfinalizer(envoy.delete)
+    envoy.commit()
+    return envoy
+
+
+@pytest.fixture(scope="module")
+def authorino_parameters(authorino_parameters, authorino_cert, create_secret):
+    """Setup TLS for authorino"""
+    authorino_secret_name = create_secret(authorino_cert, "authcert")
+    authorino_parameters["listener_certificate_secret"] = authorino_secret_name
+    yield authorino_parameters
+
+
+@pytest.fixture(scope="module")
+def auth(auth, authorization):
```

Comment:
Auth fixture has no binding to Authorization, only `client` does and since I do not use the default `client` fixture, the configuration would not be applied